src/utils/useChatHandler.ts

```diff
@@ -0,0 +1,54 @@
+import { useState } from "react"
+import { useChatHistory } from "./useChatHistory"
+import { ChatAPI } from "@/apis/chat"
+
+export const useChatHandler = (botId: string, CHAT_SESSION_ID: string) => {
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+  const [isError, setIsError] = useState<boolean>(false)
+
+  // Vu: Custom hook to manage chat history.
+  /* 
+Vu: in the old commit, I use useChatHistory directly in ChatWindow.
+- In ChatWindow, it needs chatHistory and clearChatHistory from useChatHistory
+- In this useChatHandler, it needs addMessageToChatHistory from useChatHistory
+- If I declare both useChatHistory in both ChatWindow and useChatHandler, it will create 2 separated instances in both places. Resulting an error: when user chat, it doesn't show in UI although the server responded and have new data in Session Storage.
+- That's why we include useChatHistory inside this hook, and return all useChatHistory's values so that they can be used in ChatWindow.
+  */
+  const { chatHistory, addMessageToChatHistory, clearChatHistory } =
+    useChatHistory()
+
+  // Vu: This function display new message (from message input and clickable options) to UI, also send data to server and display answer in UI
+  const sendMessageToServerAndDisplay = async (messageFrom: string) => {
+    addMessageToChatHistory(messageFrom, "user")
```

Comment:
setError(false) here

Reply:
Updated in commit [724c77b](https://github.com/AI-ChatBot-Yoga/ai-
chatbot/pull/31/commits/724c77bd0e7790013d2cffaae11d5779ec540cd1)